### PR TITLE
Feature/34 introduce unit tests for model folder

### PR DESCRIPTION
**Pull request by tmikula-dev**
Implement unit tests using pytest for model folder.

- unit tests
- for observation of code coverage by unit tests use pytest-cov.
- goal is 80+% score.

### CODE REVIEW

File: tests/model/test_consolidated_issue.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+from github.Issue import Issue
+
+from living_documentation_generator.model.consolidated_issue import ConsolidatedIssue
+from living_documentation_generator.model.project_status import ProjectStatus
+
+
+# update_with_project_data
+
+
+# THIS MIGHT NOT BE NEEDED, SINCE WE TEST ASSIGN AND APPEND
```
Comment (miroslavpojer) [inline tests/model/test_consolidated_issue.py:25]: You are right. An extra test is not needed here. The coverage will be reached from the test on a higher level.

File: tests/model/test_consolidated_issue.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# generate_page_filename
+
+
+def test_generate_page_filename_correct_behaviour():
+    mock_issue = Issue(None, None, {"number": 1, "title": None}, completed=True)
+    consolidated_issue = ConsolidatedIssue("organization/repository", mock_issue)
+
+    actual = consolidated_issue.generate_page_filename()
+
+    assert actual == "1.md"
```
Comment (miroslavpojer) [inline tests/model/test_consolidated_issue.py:45]: In this test I would expect title filled.

**Review by miroslavpojer — COMMENTED** [review]
- [ ] pull
- [ ] code review
- [ ] all run

File: tests/model/test_consolidated_issue.py (hunk 2 above)
Comment (MobiTikula) [reply tests/model/test_consolidated_issue.py]: Nice catch! Changes made in commit: [e760a98](https://github.com/AbsaOSS/living-doc-generator/pull/42/commits/e760a988275b27be90f0b64acddf8118271c7bfa).

File: tests/model/test_consolidated_issue.py (hunk 1 above)
Comment (MobiTikula) [reply tests/model/test_consolidated_issue.py]: Commit: [e760a98](https://github.com/AbsaOSS/living-doc-generator/pull/42/commits/e760a988275b27be90f0b64acddf8118271c7bfa).

**Comment by MobiTikula** [comment]
Release Notes:
- unit tests using pytest for model folder

File: tests/model/test_project_issue.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        "Priority": ["Priority1", "Priority2"],
+        "Size": ["Size1", "Size2"],
+        "MoSCoW": ["MoSCoW1", "MoSCoW2"],
+    }
+
+    actual = project_issue.loads(issue_json, project)
+
+    assert actual is not None
+    assert actual.number == issue_json["content"]["number"]
+    assert actual.organization_name == issue_json["content"]["repository"]["owner"]["login"]
```
Comment (miroslavpojer) [inline tests/model/test_project_issue.py:49]: Is this correct? login.name != orga name in GH. But maybe in Project it is different.

**Review by miroslavpojer — COMMENTED** [review]
- [ ] pull
- [ ] code review
- [ ] run local
- [ ] unit tests

File: living_documentation_generator/model/project_issue.py [hunk 4]
```diff
@@ -67,14 +67,23 @@ def loads(self, issue_json: dict, project: GithubProject) -> Optional["ProjectIs
...
-            logger.debug("No issue data provided in received json.")
-            logger.debug(issue_json)
+        if "content" not in issue_json:
+            logger.debug("No issue data provided in received json: %s.", issue_json)
             return None
 
-        self.__number = issue_json["content"]["number"]
-        self.__organization_name = issue_json["content"]["repository"]["owner"]["login"]
-        self.__repository_name = issue_json["content"]["repository"]["name"]
+        try:
```
[the remark this answers is not in the document]
Comment (MobiTikula) [reply living_documentation_generator/model/project_issue.py:74]: In this way, I thought, I get full information about, what is wrong with project issue. BUT after I saw the block of code now after some time, I do agree with you. See change in commit:

File: tests/model/test_github_project.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        "repository": {
+            "projectV2": {
+                "fields": {"nodes": [{"name": "field", "options": [{"name": "option1"}, {"name": "option2"}]}]}
+            }
+        }
+    }
+
+    actual = github_project.loads(project_json, repository, field_option_response)
+
+    assert actual.id == project_json["id"]
```
[the remark this answers is not in the document]
Comment (MobiTikula) [reply tests/model/test_github_project.py:40]: This was a wrong pattern in the whole project. Changed in commit:

File: living_documentation_generator/model/github_project.py [hunk 6]
```diff
@@ -72,17 +82,26 @@ def loads(self, project_json: dict, repository: Repository, field_option_respons
...
-        self.__organization_name = repository.owner.login
+        try:
+            self.__id = project_json["id"]
+            self.__number = project_json["number"]
+            self.__title = project_json["title"]
+            self.__organization_name = repository.owner.login
+
+            logger.debug("Updating field options for projects in repository `%s`.", repository.full_name)
+        except KeyError:
+            logger.error(
```
[the remark this answers is not in the document]
Comment (MobiTikula) [reply living_documentation_generator/model/github_project.py:93]: See the commit:

File: tests/model/test_project_issue.py (hunk 3 above)
Comment (MobiTikula) [reply tests/model/test_project_issue.py:49]: I looked again in the issue_json structure for project_issue and this should be correct. In the structure you can see the name of the the repository and after the owner of the repository behind the login value.
```
                     nodes {{
                          content {{
                              ... on Issue {{
                                title
                                state
                                number
                                repository {{
                                  name
                                  owner {{
                                    **login**
                                  }}
                                }}
                              }}
                            }}
```

File: living_documentation_generator/model/github_project.py [hunk 7]
```diff
@@ -53,6 +53,11 @@ def title(self) -> str:
         """Getter of the project title."""
         return self.__title
 
+    @title.setter
```
[the remark this answers is not in the document]
Comment (MobiTikula) [reply living_documentation_generator/model/github_project.py]: Fixture implemented see commit: [ab39950](https://github.com/AbsaOSS/living-doc-generator/pull/42/commits/ab39950fe84c66e6792e50c283c6ea86f1b2c8d3).

File: living_documentation_generator/model/project_issue.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply living_documentation_generator/model/project_issue.py:74]: In the current code, the problem still needs to be visible. The error log will print the received json but not the problematic part.

**Review by miroslavpojer — COMMENTED** [review]
- [ ] pull
- [ ] code review
- [ ] run locally
- [ ] run pytest and coverage

File: living_documentation_generator/model/project_issue.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (MobiTikula) [reply living_documentation_generator/model/project_issue.py:74]: Fixed in commit:

File: tests/conftest.py [hunk 8]
```diff
@@ -48,3 +51,18 @@ def mock_rate_limiter(mocker):
...
     yield mock_log_config
+
+
+@pytest.fixture
+def github_project_setup(mocker):
+    project = mocker.Mock(spec=GithubProject)
+    project_json = {"id": "123", "number": 1, "title": "Test Project"}
+
+    repository = mocker.Mock(spec=Repository)
+    repository.owner.login = "organizationABC"
```
[the remark this answers is not in the document]
Comment (MobiTikula) [reply tests/conftest.py:62]: After discussion, this is the correct value of the field.

File: tests/model/test_project_issue.py [hunk 9]
```diff
@@ -95,30 +74,7 @@ def test_loads_with_incorrect_json_structure_for_repository_name(mocker):
...
-    assert actual.number == 1
-    assert actual.repository_name == "repositoryABC"
-    assert actual.organization_name == ""
-    mock_log.debug.assert_called_once_with(
-        "Wrong project issue json structure for `organization_name` value: %s.", incorrect_json
-    )
+    assert 1 == actual.number
+    assert "" == actual.organization_name
+    assert "" == actual.repository_name
+    mock_log.debug.assert_called_once_with("KeyError occurred while parsing issue json: %s.", incorrect_json)
```
[the remark this answers is not in the document]
Comment (MobiTikula) [reply tests/model/test_project_issue.py]: See commit: [0ebe871](https://github.com/AbsaOSS/living-doc-generator/pull/42/commits/0ebe871392693b456c134964659894d672aea50f).

**Review by miroslavpojer — APPROVED** [review]

**Pull request merged**